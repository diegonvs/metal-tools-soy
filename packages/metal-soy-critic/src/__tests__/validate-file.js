const {getFixturePath} = require('../../test/util');
const validateFile = require('../validate-file');

describe('Validate file', () => {
  function validate(fileName) {
    return validateFile(getFixturePath(fileName));
  }

  test('should pass', () => validate('Test.soy').then(result => {
      expect(result.status).toBe(true);
  }));

  test('should fail; missing params', () => validate('MissingParams.soy').then(result => {
      expect(result.status).toBe(false);
      expect(result.messages).toMatchSnapshot();
  }));

  test('should fail; missing internal', () => validate('MissingInternal.soy').then(result => {
      expect(result.status).toBe(false);
      expect(result.messages).toMatchSnapshot();
  }));

  test('should fail; missing import', () => validate('MissingCallImport.soy').then(result => {
      expect(result.status).toBe(false);
      expect(result.messages).toMatchSnapshot();
  }));

  test('should fail; missing required', () => validate('MissingRequired.soy').then(result => {
      expect(result.status).toBe(false);
      expect(result.messages).toMatchSnapshot();
  }));

  test('should fail; invalid namespace', () => validate('InvalidNamespace.soy').then(result => {
      expect(result.status).toBe(false);
      expect(result.messages).toMatchSnapshot();
  }));
});
