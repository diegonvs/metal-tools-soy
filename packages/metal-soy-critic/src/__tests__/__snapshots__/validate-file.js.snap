// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate file should fail; missing import 1`] = `
Array [
  "It looks like the following component calls are missing an [33mimport[39m:

[31mMyExternalComponent[39m",
]
`;

exports[`Validate file should fail; missing internal 1`] = `
Array [
  "Based on their name, these attributes should have the [33m.internal()[39m config added:

[31mcount_[39m",
]
`;

exports[`Validate file should fail; missing params 1`] = `
Array [
  "The following params should be included in [33mSTATE[39m:

[31moptionalInfo[39m",
]
`;
